rust/main/hyperlane-base/src/settings/chains.rs

```diff
@@ -978,6 +985,11 @@ fn build_sealevel_rpc_client(
     let middleware_metrics = chain_conf.metrics_conf();
     let rpc_client_url = connection_conf.url.clone();
     let client_metrics = metrics.client_metrics();
+
+    // increment provider metric count
+    let chain_name = PrometheusConfig::chain_name(&middleware_metrics.chain);
```

Comment:
Not sure if we should increment this when we create an RPC Client or when we create an actual **Provider** struct. I think its usually 1:1 anyways though.

Reply:
it'd be much cleaner to increment in the provider constructor. We want new code that creates providers to automatically increment this metric, rather than needing it done manually

Reply:
yeah, that makes sense